src/renderers/webxr/WebXRManager.js

```diff
@@ -662,13 +686,11 @@ class WebXRManager extends EventDispatcher {
 
 			//
 
-			const inputSources = session.inputSources;
-
 			for ( let i = 0; i < controllers.length; i ++ ) {
```

Comment:
Wouldn't it be better to just do:

```js
for ( let i = 0; i < inputSources.length; i ++ ) {
```

Reply:
This is supposed to guarantue that the controller.update event is called with the input source that was previously associated with it (in the onInputSources changed events where we call onConnect and onDisconnect). And session.inputSources does not necessarily be in the same order if I'm not mistaken (e.g. when using multitouch)

Reply:
Hmm, what't the purpose of `if ( i >= inputSources.length ) continue;` then?

Reply:
Input sources are added to this array when connected to controllers for the first time. Ah ok so you're suggesting to use that to loop through, yeah that makes sense :)

https://github.com/mrdoob/three.js/blob/3414e135c1e520cfb01a43ab251fcc6c6147af37/src/renderers/webxr/WebXRManager.js#L392

Reply:
Seems like the code is a bit hard to follow then...

Reply:
Ok! Do you think it would be easier if the inputSource array was always the same length as controllers?
Or should I better use the inputSourceMap (I was wondering why a map was used here, do you know if there's any specific reason for that? Having two arrays with a direct mapping seems easier to me as there would probably never be that many controllers anyways)

Reply:
Hey, any news for this? :)

Reply:
Sorry, the last few months have been a bit hectic... 😅

Reply:
No problem, thanks for reviewing now!